website/docs/Contribute.md

```diff
@@ -120,7 +120,23 @@ Tests are automatically run via GitHub actions. There are two workflows:
 1. [build.yml](https://github.com/microsoft/autogen/blob/main/.github/workflows/build.yml)
 1. [openai.yml](https://github.com/microsoft/autogen/blob/main/.github/workflows/openai.yml)
 
-The first workflow is required to pass for all PRs. The second workflow is required for changes that affect the openai tests. The second workflow requires approval to run. When writing tests that require openai, please use [`pytest.mark.skipif`](https://github.com/microsoft/autogen/blob/main/test/test_client.py#L13) to make them run in one python version only when openai is installed. If additional dependency for this test is required, install the dependency in the corresponding python version in [openai.yml](https://github.com/microsoft/autogen/blob/main/.github/workflows/openai.yml).
+The first workflow is required to pass for all PRs (and it doesn't do any OpenAI calls). The second workflow is required for changes that affect the OpenAI tests (and does actually call LLM). The second workflow requires approval to run. When writing tests that require OpenAI calls, please use [`pytest.mark.skipif`](https://github.com/microsoft/autogen/blob/b1adac515931bf236ac59224269eeec683a162ba/test/oai/test_client.py#L19) to make them run in one python version only when `openai` package is installed. If additional dependency for this test is required, install the dependency in the corresponding python version in [openai.yml](https://github.com/microsoft/autogen/blob/main/.github/workflows/openai.yml).
+
+#### Run non-OpenAI tests
+
+To run the subset of the tests not depending on `openai` (and not calling LLMs)):
+- Create an empty file `OAI_CONFIG_LIST`, add `[]` (make it a valid JSON with empty collection)
+- Install `pytest`
+- Remove `openai` library, this is required to skip the tests which check for `openai` presence ([`pytest.mark.skipif`](https://github.com/microsoft/autogen/blob/b1adac515931bf236ac59224269eeec683a162ba/test/oai/test_client.py#L19))
+- Run the tests from the `test` folder. Make sure they all pass, this is required for [build.yml](https://github.com/microsoft/autogen/blob/main/.github/workflows/build.yml) checks to pass
+- Reinstall `openai`
```

Comment:
In future, we can put all the tests requiring openai in one place and skip/run them in the pytest command in the workflow, instead of skipping each test based on whether openai is installed.

Reply:
There're 10 files that use the pattern of `skip if openai or some other dependencies can't` be imported

<img width="350" alt="image" src="https://github.com/microsoft/autogen/assets/7947027/bdae8c29-dc2e-4a52-abd1-d9c2e25f66b3">

To make an incremental change we can go with adding a support for skip flag, e.g. `pytest test --skip-openai`.

1. Give pytest a global config via `test/conftest.py`

 ``` py
 import pytest

skip_openai = False


def pytest_addoption(parser):
    parser.addoption("--skip-openai", action="store_true", help="Skip all tests that require openai")


@pytest.fixture(scope='session')
def get_skip_openai(request):
    return request.config.getoption("--skip-openai", False)

@pytest.hookimpl(tryfirst=True)
def pytest_configure(config):
    global skip_openai
    skip_openai = config.getoption("--skip-openai", False)
```

2. Extend the skip criteria everywhere there's the import/skip pattern, e.g. `test/test_retrieve_utils.py`
```py
import pytest
from conftest import skip_openai

try:
    import chromadb
    from autogen.retrieve_utils import (
        split_text_to_chunks,
        extract_text_from_pdf,
        split_files_to_chunks,
        get_files_from_dir,
        is_url,
        create_vector_db_from_dir,
        query_vector_db,
    )
    from autogen.token_count_utils import count_token
except ImportError:
    skip = True
else:
    skip = False or skip_openai
    
    ...
```

3. Update `build.yaml`
```yaml
      - name: Test with pytest
        if: matrix.python-version != '3.10'
        run: |
          pytest test --skip-openai
```

Reply:
Adding this `skip-openai` option sounds good to me!